qemu/tests/block_with_iommu.py

```diff
@@ -60,6 +61,11 @@ def verify_iommu_enabled():
     session = vm.wait_for_login(timeout=360)
     verify_iommu_enabled()
 
+    session.cmd(params.get("dd_cmd"))
+
     if params.get("reload_kernel_cmd"):
         reload_kernel(session)
-        vm.reboot(session)
+
+    vm.reboot(session)
+    vm.wait_for_login(timeout=360)
+    session.cmd(params.get("dd_cmd"))
```

Comment:
```suggestion
    session = vm.reboot(session, timeout=360)
    session.cmd(params.get("dd_cmd"))
```
Actually, it will return a new session for the guest after `vm.reboot`, so I think we do not need to `vm.wait_for_login` again.
Please correct me if I misunderstood, thanks.
BTW, I am curious why the line 71 `session.cmd(params.get("dd_cmd"))` did not raise error here since the session object will be closed in the `vm.reboot `
Reference codes:
https://github.com/avocado-framework/avocado-vt/blob/ff99a218cdacd481f8ce1da634b8111d26cc9814/virttest/qemu_vm.py#L5640-L5642
https://github.com/avocado-framework/avocado-vt/blob/ff99a218cdacd481f8ce1da634b8111d26cc9814/virttest/qemu_vm.py#L5625-L5627

Reply:
i think it session=vm.wait_for_login(timeout=360) ? wait_for_login is need not only run command but also we need to check the guest is bootable, and some cmd need to wait guest login able.

Reply:
> i think it session=vm.wait_for_login(timeout=360) ? wait_for_login is need not only run command but also we need to check the guest is bootable, and some cmd need to wait guest login able.

Hi @qingwangrh 
Actually, it has been logged in the guest in `vm.reboot` to check if it is bootable.

https://github.com/avocado-framework/avocado-vt/blob/ff99a218cdacd481f8ce1da634b8111d26cc9814/virttest/qemu_vm.py#L5640-L5642

Reply:
@YongxueHong updated !